### PR TITLE
add skip mysqldump config

### PR DESCRIPTION
**Pull request by miniframework**

### CODE REVIEW

File: canal/config.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type Config struct {
 	Charset  string `toml:"charset"`
 	ServerID uint32 `toml:"server_id"`
 	Flavor   string `toml:"flavor"`
+    SkipMysqlDump bool `toml:mysqldump`
```
Comment (siddontang) [inline canal/config.go]: please format your code.

File: canal/config.go [hunk 2]
```diff
@@ -76,6 +77,7 @@ func NewDefaultConfig() *Config {
 	c.ServerID = uint32(rand.Intn(1000)) + 1001
 
 	c.Flavor = "mysql"
+    c.SkipMysqlDump = true
```
Comment (siddontang) [inline canal/config.go]: ditto

File: canal/canal.go [hunk 3]
```diff
@@ -59,10 +59,12 @@ func NewCanal(cfg *Config) (*Canal, error) {
 	c.master = &masterInfo{}
 
 	var err error
-
-	if err = c.prepareDumper(); err != nil {
-		return nil, errors.Trace(err)
-	}
+    //skip mysqldump bin, only use binlog 
+    if c.cfg.SkipMysqlDump != true {
```
Comment (siddontang) [inline canal/canal.go]: make alignment.

File: canal/canal.go [hunk 4]
```diff
@@ -128,6 +130,7 @@ func (c *Canal) prepareDumper() error {
 func (c *Canal) Start() error {
 	c.wg.Add(1)
 	go c.run()
+	c.wg.Wait()
```
Comment (siddontang) [inline canal/canal.go]: This will block the start, I don't think we need this feature in this PR.

File: canal/canal.go [hunk 5]
```diff
@@ -59,9 +59,11 @@ func NewCanal(cfg *Config) (*Canal, error) {
 	c.master = &masterInfo{}
 
 	var err error
-
-	if err = c.prepareDumper(); err != nil {
-		return nil, errors.Trace(err)
+	//skip mysqldump bin, only use binlog 
+	if c.cfg.SkipMysqlDump != true {
```
Comment (siddontang) [inline canal/canal.go]: if !c.cfg.SkipMysqlDump

**Comment by siddontang** [comment]
CI failed @miniframework

File: canal/config.go [hunk 6]
```diff
@@ -76,6 +77,7 @@ func NewDefaultConfig() *Config {
 	c.ServerID = uint32(rand.Intn(1000)) + 1001
 
 	c.Flavor = "mysql"
+	c.SkipMysqlDump = true
```
Comment (NearTan) [inline canal/config.go]: I don't think set default value as `true` here is a good idea. 

btw: I suggest use the consistent case word `MySQL`.

**Comment by siddontang** [comment]
No update, I will close it now. 

May re-open later.

**Pull request closed without merging**